### DISCUSSION TITLE
Editable cell referenced by ajax blur event update not moving focus properly with tab key second time tabbing through

### DISCUSSION THREAD

**Discussion by BryceMcGilly** [primefaces/community#3196] [PrimeFaces] [answered]
Hello,

I have an editable datatable similar to the 'Cell Editing with Click and RightClick' example in the Primefaces showcase. I have a row of cells A, B, C, D, E, F, and G. Cell B has a <p:ajax event="blur" process="@this" update="componentInC"/>.  The first time tabbing through the cells, it tabs through in correct order from one cell to the other.  However, if I left click back on cell A and tab through the cells a second time, it will go from A, B, C, then skip to G.  If I remove the update="componentInC" in the ajax event="blur", the problem does not occur and it will tab through the cells correctly the second time tabbing through them.

It looks like the first update does not mess up the tabbing order, but the second update does.

For reference I'm using Primefaces 15.0.0-RC1
This was also reproduced in Primefaces 14.0.9
Also, I'm using the Google Chrome browser on Windows 11.

Thanks :)

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

can you create a small reproducer for us to debug the issue?

**Reply by melloware** [reply]
Also @BryceMcGilly not sure why you are using an AJAX `blur` on the cell when that is how cell editing works its has its own `onBlur` for leaving the cell.

See: https://primefaces.github.io/primefaces/vdldoc/p/dataTable.html#saveOnCellBlur

By default it calls onBlur so now you have 2 `blur` events firing at the same time and both trying to update the cell so i am sure there is a conflict.

**Comment by BryceMcGilly** [comment]
Thanks for the response @melloware 

Here is the reproducer as you requested: https://github.com/BryceMcGilly/primefaces-test-tabbing-issue

I've tried to simplify the issue.  To reproduce the behavior:

1) Increment the Quantity of the Thriller item to 1
2) Press tab to tab through the cells in the rows, notice the tabbing order is correct
3) Increment the Quantity of the Thriller item again to 2
4) Press tab to tab through the cells in the row again, notice the tabbing order is messed up after trying to tab out of the "Minimum Order Quantity" cell to the next cell
5) The order remains messed up if you left click on the Thriller cell and try to tab through the table
6) Increment the "Minmum Order Quantity" cell of the Thriller row to 1 and try to tab through the table starting with the Thriller cell, notice the Minimum Order Quantity cell is now skipped. Also notice once it's skipped over, the spinner in this cell is now frozen and you cannot increment or decrement the Minimum Order Quantity.

Thank you,
Bryce

**Reply by melloware** [reply]
OK can you open a GitHub Issue ticket please it looks like its Spinner related: https://github.com/primefaces/primefaces/issues

**Reply by BryceMcGilly** [reply]
Created an issue:  https://github.com/primefaces/primefaces/issues/13099

**Reply by melloware** [reply, marked as answer]
Commented in the ticket: https://github.com/primefaces/primefaces/issues/13099
